src/Features/Core/Portable/DesignerAttributes/AbstractDesignerAttributeService.cs

```diff
@@ -17,13 +17,9 @@ internal abstract class AbstractDesignerAttributeService : IDesignerAttributeSer
         // on remote host, so we need to make sure given input always belong to right workspace where
         // the session belong to.
         private readonly Workspace _workspace;
-        private readonly SemaphoreSlim _gate;
-
-        private KeepAliveSession _sessionDoNotAccessDirectly;
```

Comment:
Are all uses of KeepAliveSession going away? Why are the ones that are left still left?

Reply:
people who want to use OOP as stateful service keep connection open. example will be nuget indexer. it maintain state on server side.

Reply:
@heejaechang Should we rename it to StatefulSession then?

Reply:
not in this PR :)

Reply:
but added that to this work item https://github.com/dotnet/roslyn/issues/24836